theme/js/components/pagination/pagination.vue

```diff
@@ -52,55 +51,84 @@ export default {
-        >{{ index }}</a
-      >
-      <span class="ellipsis" role="img" aria-label="ellipsis" aria-hidden="true" v-else>...</span>
-    </li>
-    <li>
-      <a
-        :class="{ active: page === pages.length }"
-        :aria-disabled="page === pages.length"
-        @click.prevent="_onClick(pages.length)"
-        >{{ pages.length }}</a
-      >
-    </li>
-    <li>
-      <a
-        :class="{ disabled: page === pages.length }"
-        class="next"
-        :aria-disabled="page === pages.length"
-        @click.prevent="_onClick(page + 1)"
-      ></a>
-    </li>
-  </ul>
+  <nav role="navigation" class="fr-pagination fr-pagination--centered" aria-label="Pagination">
+    <ul class="fr-pagination__list">
+      <li>
+        <a
+          :href="page === 1 ? null :'#'"
+          class="fr-pagination__link fr-pagination__link--first"
+          @click.prevent="_onClick(1)"
+        >
+          {{ $t('First page') }}
+        </a>
+      </li>
+      <li>
+        <a
+          :href="page === 1 ? null : '#'"
+          class="fr-pagination__link fr-pagination__link--prev fr-pagination__link--lg-label"
+          @click.prevent="_onClick(page - 1)"
+        >
+          {{ $t('Previous page') }}
+        </a>
+      </li>
+      <li>
+        <a
+          :aria-current="page === 1 ? 'page' : null"
+          :href="page === 1 ? null : '#'"
+          class="fr-pagination__link fr-displayed-sm"
+          :title="$t('Page', {nb: 1})"
+          @click.prevent="_onClick(1)"
+        >
+          1
+        </a>
+      </li>
+      <li v-for="index in visiblePages">
+        <a
+          class="fr-pagination__link"
+          :class="{'fr-displayed-lg': index < page - 1 || index > page + 1}"
+          :aria-current="page === index ? 'page' : null"
+          :href="page === index ? null : '#'"
+          :title="$t('Page', {nb: index})"
+          @click.prevent="_onClick(index)"
+          v-if="index"
+          >
+          {{ index }}
+        </a>
+        <a class="fr-pagination__link fr-displayed-lg" v-else>
+          …
+        </a>
+      </li>
+      <li>
+        <a
+          class="fr-pagination__link"
+          :aria-current="page === pages.length ? 'page' : null"
+          :href="page === pages.length ? null : '#'"
+          :title="$t('Page', {nb: pages.length})"
+          @click.prevent="_onClick(pages.length)"
+        >
+          {{ pages.length }}
+        </a>
+      </li>
+      <li>
+        <a
+          class="fr-pagination__link fr-pagination__link--next fr-pagination__link--lg-label"
+          :href="page === pages.length ? null : '#'"
+          @click.prevent="_onClick(page + 1)"
+        >
+          {{ $t('Next page') }}
+        </a>
+      </li>
+      <li>
+        <a
+          class="fr-pagination__link fr-pagination__link--last"
+          :href="page === pages.length ? null : '#'"
+          @click.prevent="_onClick(pages.length)"
+        >
+          {{ $t('Last page') }}
```

Comment:
It's here for accessibility. We can add the `fr-pagination__link--lg-label` class if we want to show it on large screen but we'll have to reduce the number of pages shown because of some edge cases at the start of the lg media query.